### PR TITLE
feat(dropdown): style for empty state and arrow rotate

### PR DESCRIPTION
**Pull request by kdabo**
**Reviewers:@endritbajo @beerecca @katia 

Changes:
- added style to rotate arrow on drop-down open
- added empty state style
- adjust the documentation style

### CODE REVIEW

File: src/components/_dropdown.scss [hunk 1]
```diff
@@ -87,6 +93,11 @@
     .#{$ns}dropdown__menu {
       display: block;
     }
+
+    .#{$ns}icon--up {
+      transform: rotate(180deg);
+      top: 30%;
```
Comment (kdabo) [inline src/components/_dropdown.scss]: not really happy with this solution, but if I don't add `top: 30%` the arrow is not centered.
do you guys have any idea why?

Comment (endritbajo) [inline src/components/_dropdown.scss]: When you're adding the transform you're actually overriding the default `transform: translateY(-50%)`.
So you need to replace this with `transform: translateY(-50%) rotate(180deg)`

File: src/components/_dropdown.scss [hunk 2]
```diff
@@ -12,9 +12,10 @@
 //        <button class="sb-input">
 //          Choice 1
 //        </button>
-//        <svg class="sb-icon" viewBox="-5 -5 16 16"><path d="M0 0l3.5 4L7 0H0"></svg>
+//        <svg class="sb-icon sb-icon--up" viewBox="-5 -5 16 16"><path d="M0 0l3.5 4L7 0H0"></svg>
```
Comment (endritbajo) [inline src/components/_dropdown.scss]: I would name this `sb-icon--flip` so we can actually reuse it in the future in other components Ex. table headers, etc.

Comment (endritbajo) [inline src/components/_dropdown.scss]: Also, I would this class dynamically only when we want it to rotate and not being by default there.

Comment (kdabo) [inline src/components/_dropdown.scss]: Ok. I added it there for the sake of preview in the documentation.

File: src/components/_dropdown.scss [hunk 3]
```diff
@@ -15,6 +15,7 @@
 //        <svg class="sb-icon" viewBox="-5 -5 16 16"><path d="M0 0l3.5 4L7 0H0"></svg>
 //      </div>
 //      <ul class="sb-dropdown__menu" role="listbox">
+//        <li class="sb-dropdown__menu-item sb-dropdown__menu-item--disabled" tabindex="0" role="option">No items to select</li>
```
Comment (endritbajo) [inline src/components/_dropdown.scss]: Update the `tabindex="-1"` as we don't want it to be focusable

Comment (endritbajo) [inline src/components/_dropdown.scss]: Maybe we can put it in a subsection as 'No options dropdown'

File: src/components/_dropdown.scss [hunk 4]
```diff
@@ -61,7 +62,6 @@
   }
 
   &__menu-item {
-    padding: sb-px2rems(11px);
```
Comment (beerecca) [inline src/components/_dropdown.scss]: Why no padding anymore?

File: src/components/_dropdown.scss [hunk 5]
```diff
@@ -58,6 +59,7 @@
     margin: -1px 0 0 0;
     max-height: 300px;
     overflow-y: scroll;
+    z-index: 100;
```
Comment (endritbajo) [inline src/components/_dropdown.scss]: Is this number for a reason or is it random? We should set it less than the `toast` z-index.

File: src/components/_dropdown.scss [hunk 6]
```diff
@@ -138,4 +155,23 @@
 //  </div>
 //
 // Styleguide dropdown.validation
+
+// No options dropdown
```
Comment (endritbajo) [inline src/components/_dropdown.scss]: We decided to remove the `No options` li menu item right?

**Pull request merged**